.github/workflows/ci.yml

```diff
@@ -46,9 +47,82 @@ jobs:
           PG_OPERATOR_VERSION: 2.5.0
           PERCONA_VERSION_SERVICE_URL: https://check-dev.percona.com/versions/v1
 
+  integration-tests:
+    name: Integration tests
+    timeout-minutes: 30
+    strategy:
+      fail-fast: true
+    runs-on: ubuntu-latest
+    env:
+      TOOLS_PATH: "/opt/tools/bin"
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v4
+
+      - name: Set up Go release
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          check-latest: true
+
+      - name: Build operator
+        run:
+          make build
+
+      - name: Install tools
+        run: |
+          mkdir -p $TOOLS_PATH
+          echo $TOOLS_PATH >> $GITHUB_PATH
+          # Kubectl
+          curl -LO https://dl.k8s.io/release/v$KUBECTL_VERSION/bin/linux/amd64/kubectl
+          curl -LO "https://dl.k8s.io/v$KUBECTL_VERSION/bin/linux/amd64/kubectl.sha256"
+          echo "$(cat kubectl.sha256)  kubectl" | sha256sum --check
+          chmod +x kubectl
+          cp kubectl "$TOOLS_PATH"
+          rm kubectl
+          # Kuttl
+          curl -Lo ./kubectl-kuttl https://github.com/kudobuilder/kuttl/releases/download/v0.15.0/kubectl-kuttl_0.15.0_linux_x86_64
+          chmod +x ./kubectl-kuttl
+          mv ./kubectl-kuttl "$TOOLS_PATH"
+        env:
+          KUBECTL_VERSION: 1.27.9
+
+      - name: Run core tests
+        if: always()
+        run: |
+          kubectl kuttl test --config ./tests/integration/kuttl-core.yaml
+        env:
+          PXC_OPERATOR_VERSION: 1.15.0
+          PSMDB_OPERATOR_VERSION: 1.18.0
+          PG_OPERATOR_VERSION: 2.5.0
+          PERCONA_VERSION_SERVICE_URL: https://check-dev.percona.com/versions/v1
+
+      - name: Run features tests
+        if: always()
+        run: |
+          kubectl kuttl test --config ./tests/integration/kuttl-features.yaml
+        env:
+          PXC_OPERATOR_VERSION: 1.15.0
+          PSMDB_OPERATOR_VERSION: 1.18.0
+          PG_OPERATOR_VERSION: 2.5.0
+          PERCONA_VERSION_SERVICE_URL: https://check-dev.percona.com/versions/v1
+
+      - name: Run upgrade tests
+        if: always()
+        run: |
+          kubectl kuttl test --config ./tests/integration/kuttl-upgrade.yaml
+        env:
+          PXC_OPERATOR_VERSION: 1.15.0
+          PSMDB_OPERATOR_VERSION: 1.18.0
+          PG_OPERATOR_VERSION: 2.5.0
+          PREVIOUS_PG_OPERATOR_VERSION: 2.4.1
+          PREVIOUS_PXC_OPERATOR_VERSION: 1.14.0
+          PREVIOUS_PSMDB_OPERATOR_VERSION: 1.17.0
+          PERCONA_VERSION_SERVICE_URL: https://check-dev.percona.com/versions/v1
 
   eks-e2e-tests:
     name: E2E tests (EKS)
+    if: contains(github.event.pull_request.labels.*.name, 'test/e2e') || contains(github.event.pull_request.labels.*.name, 'test/e2e-upgrade') || contains(github.event.pull_request.labels.*.name, 'test/e2e-core')
```

Comment:
The E2E step is a required check in CI, so we skip it, the PR will be blocked, wouldn't it? Should we configure Merge Gatekeeper then?

Reply:
Although now it looks like the PR is not blocked by the skipped step. Not sure😅

Reply:
I was concerned about this as well but I didn't check how the merge gatekeeper works internally because I also saw that the PR wasn't blocked by it. It seems gatekeeper is smart enough to know that a skipped job should not be taken into account.